plugins/outputs/http/http.go

```diff
@@ -0,0 +1,182 @@
+  url = "http://127.0.0.1:8080/metric"
+  ## HTTP Content-Type. Default : application/json
+  content_type = "application/json"
+  ## Set the number of times to retry when the status code is not 200 or an error occurs during HTTP call. Default: 3
+  retry = 3
+  ## Configure TLS handshake timeout. Default : 10
+  tls_handshake_timeout = 10
+  ## Configure response header timeout in seconds. Default : 3
+  response_header_timeout = 3
+  ## Configure dial timeout in seconds. Default : 3
+  dial_timeout = 3
+  ## Configure HTTP Keep-Alive. Default : 0
+  keepalive = 0
+  ## Configure HTTP expect continue timeout in seconds. Default : 0
+  expect_continue_timeout = 3
+  ## Configure idle connection timeout in seconds. Default : 0
+  idle_conn_timeout = 3
+
+  ## Data format to output.
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  data_format = "influx"
+`
+
+const (
+	POST         = "POST"
+	CONTENT_TYPE = "Content-Type"
+
+	DEFAULT_RETRY                   = 3
+	DEFAULT_CONTENT_TYPE            = "application/json"
+	DEFAULT_TLS_HANDSHAKE_TIMEOUT   = 10
+	DEFAULT_RESPONSE_HEADER_TIMEOUT = 3
+	DEFAULT_DIAL_TIME_OUT           = 3
+	DEFAULT_KEEP_ALIVE              = 3
+	DEFAULT_EXPECT_CONTINUE_TIMEOUT = 3
+	DEFAULT_IDLE_CONN_TIMEOUT       = 3
+)
+
+type Http struct {
+	// http required option
+	URL string `toml:"url"`
+
+	// http default value가 있는 option
+	ContentType           string `toml:"content_type"`
+	Retry                 int    `toml:"retry"`
+	TLSHandshakeTimeout   int    `toml:"tls_handshake_timeout"`
+	ResponseHeaderTimeout int    `toml:"response_header_timeout"`
+	DialTimeOut           int    `toml:"dial_timeout"`
+	KeepAlive             int    `toml:"keepalive"`
+	ExpectContinueTimeout int    `toml:"expect_continue_timeout"`
+	IdleConnTimeout       int    `toml:"idle_conn_timeout"`
+
+	client     http.Client
+	serializer serializers.Serializer
+}
+
+func (h *Http) SetSerializer(serializer serializers.Serializer) {
+	h.serializer = serializer
+}
+
+// Connect to the Output
+func (h Http) Connect() error {
+	h.client = http.Client{
+		Transport: &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout:   time.Duration(h.DialTimeOut) * time.Second,
+				KeepAlive: time.Duration(h.KeepAlive) * time.Second,
+			}).Dial,
+			TLSHandshakeTimeout:   time.Duration(h.TLSHandshakeTimeout) * time.Second,
+			ResponseHeaderTimeout: time.Duration(h.ResponseHeaderTimeout) * time.Second,
+			ExpectContinueTimeout: time.Duration(h.ExpectContinueTimeout) * time.Second,
+			IdleConnTimeout:       time.Duration(h.IdleConnTimeout) * time.Second,
+		},
+	}
+
+	return nil
+}
+
+// Close is not implemented. Because http.Client not provided connection close policy. Instead, uses the response.Body.Close() pattern.
+func (h Http) Close() error {
```

Comment:
that's my fault. In addition to this section, I will modify the functions that are implemented as non-pointer elsewhere.

Thanks